pkg/registry/node/strategy.go

```diff
@@ -153,16 +153,23 @@ func ResourceLocation(getter ResourceGetter, connection client.ConnectionInfoGet
 	}
 	host := hostIP.String()
 
-	if portReq == "" || strconv.Itoa(ports.KubeletPort) == portReq {
+	if portReq == "" || (node.Status.DaemonEndpoints.KubeletEndpoint.Port != 0 && strconv.Itoa(node.Status.DaemonEndpoints.KubeletEndpoint.Port) == portReq) ||
```

Comment:
a) This line is really long. I had to scroll the window to read the code.
b) Add a comment about what this is supposed to do and why.